Description: Drop support for rsa1
 The SSH probe currently can't run because the rsa1 type is unavailable. This
 patch removes rsa1 and adds support for ecdsa instead.
Author: Avinash H. Duduskar <strykar@hotmail.com>
Origin: Upstream
Bug-Debian: https://bugs.debian.org/962225
Applied-Upstream: https://github.com/oetiker/SmokePing/commit/62ac9fda04b994bbf4f97d3dd1cf8b92cf279e71

---
--- smokeping-2.7.3.orig/lib/Smokeping/probes/SSH.pm
+++ smokeping-2.7.3/lib/Smokeping/probes/SSH.pm
@@ -55,7 +55,7 @@ sub new($$$)
     # no need for this if we run as a cgi
     unless ( $ENV{SERVER_SOFTWARE} ) {
         
-        my $call = "$self->{properties}{binary} -t dsa,rsa,rsa1 127.0.0.1";
+        my $call = "$self->{properties}{binary} -t dsa,rsa,ecdsa 127.0.0.1";
         my $return = `$call 2>&1`;
         if ($return =~ m/$ssh_re/s){
             print "### parsing ssh-keyscan output...OK\n";
@@ -130,7 +130,7 @@ sub targetvars {
         return $class->_makevars($class->SUPER::targetvars, {
            keytype => {
                _doc => "Type of key, used in ssh-keyscan -t I<keytype>",
-	       _re => "[dr]sa1*",
+	       _re => "[ecdr]sa*",
                _example => 'dsa',
                _default => 'rsa',
            },
